backend/src/main/java/gov/cdc/usds/simplereport/service/ApiUserService.java

```diff
@@ -554,6 +570,16 @@ private ApiUser getCurrentApiUserNoCache() {
     return nonOktaUser.orElseGet(() -> getCurrentApiUserFromIdentity(userIdentity));
   }
 
+  public UserInfo getCurrentUserInfoForWhoAmI() {
```

Comment:
⚠️ this is nearly identical (minus the feature flag conditional on lines `577-579`) as the method [getCurrentUserInfo()](https://github.com/CDCgov/prime-simplereport/pull/7856/files#diff-3a4cd84e094d22aa2350c1d2c5c0a1a66e1a4952f73b13ef86654c7e2d285477R583) on lines `583-588`.

This is because `getCurrentUserInfo` was being called for, I think, at the beginning of [every graphQL request](https://github.com/CDCgov/prime-simplereport/blob/268ddb50c86572fc83419243cc36a45dfd4845a1/backend/src/main/java/gov/cdc/usds/simplereport/logging/GraphQlInterceptor.java#L58). To avoid doing an update on every request, I created a separate method for `whoami` calls

Reply:
wonder if another way to solve this is to move the db update into a lower-level call like `getCurrentApiUserNoCache` so that the db update only happens when there's a cache miss in the context holder. I'm guessing that this is where we'll put the logic for seeding new for seeding those tables when there isn't someone in Okta, so might be good to start breaking those flows out here. 

Think the benefits are 1) that way we won't have two slightly different methods that we'd have maintain separately and 2) separating these operations (getting the user, updating their records in the db) into two separate levels of abstraction (since the update db seems to be a lower-level call on the order of handling a cache miss rather than at this higher repo-orchestration level).

Reply:
Based on how frequently the debugger hits that method when I'm testing this out locally, I'm not sure I would want to move the update to `getCurrentApiUserNoCache` method. For example, we can potentially call it when we call [`getCurrentApiUser`](https://github.com/CDCgov/prime-simplereport/blob/main/backend/src/main/java/gov/cdc/usds/simplereport/service/ApiUserService.java#L532) and as a result [`getCurrentApiUserInContainedTransaction`](https://github.com/CDCgov/prime-simplereport/blob/main/backend/src/main/java/gov/cdc/usds/simplereport/service/ApiUserService.java#L381). Please let me know if I'm missing something or if you think this behavior will be different in a non-local env!! 👀

Reply:
Is `getCurrentUserInfoForWhoAmI` temporary until the entire Okta migration is complete? I'm assuming that we'll no longer need this separate method once we get to the final stage of entirely relying on our tables for auth data and updating those tables through the appropriate manage user flows

Reply:
@mpbrown Yes, it is temporary!